.github/workflows/code-gen.yaml

```diff
@@ -0,0 +1,41 @@
+name: Code Generation
```

Comment:
nit: let's rename this file to `codegen.yaml` so we don't have both `codegen` and `code-gen` ?